### PR TITLE
request logger metadata lookup

### PR DESCRIPTION
**Pull request by ryandawsonuk**
for https://github.com/SeldonIO/seldon-core/issues/3182

### CODE REVIEW

File: components/seldon-request-logger/testing/create_dummy_metadata.py
Comment (ryandawsonuk) [inline components/seldon-request-logger/testing/create_dummy_metadata.py:55]: Now works following https://github.com/SeldonIO/seldon-deploy-sdk/pull/20

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **ryandawsonuk**
You can assign the PR to them by writing `/assign @ryandawsonuk` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@ryandawsonuk: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | bff1250db44d1616dd54994d357bda7c82fa5b13 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/integration-gd6xz) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: components/seldon-request-logger/app/log_mapping.py
Comment (SachinVarghese) [inline components/seldon-request-logger/app/log_mapping.py:179]: Here selecting the first model is a key logical piece that implies this will only work with predictors with one model otherwise might give undesired results using wrong model-level metadata. There should be a check for this.

Comment (ryandawsonuk) [reply components/seldon-request-logger/app/log_mapping.py:182]: Good point, just pushed a change for this
https://github.com/SeldonIO/seldon-core/pull/3183/commits/9a9a0c86f5f875c5e48f126d26a80733fead5c0b

Comment (SachinVarghese) [reply components/seldon-request-logger/app/log_mapping.py:182]: Oh okay, maybe I wasn't clear. I meant if the number of models is greater than 1 then the function should return None. 
So in this case it doesn't make sense to use the first model's metadata as it would be considerably different from the predictor level metadata which should create the elastic index mapping. So use the metadata only if the number of models is exactly equal to one.

**Review by SachinVarghese — APPROVED** [review]
/lgtm

**Comment by ryandawsonuk** [comment]
Hmm, maybe operator tests are failing now because I rebased from master this morning

**Pull request merged**
